### PR TITLE
Cloud Builder integration for tests, builds, deployment

### PR DESCRIPTION
**Pull request by nkashy1**
**DEPENDS ON:** https://github.com/doc-ai/tensorio-models/pull/28

### CODE REVIEW

**Comment by nkashy1** [comment]
This PR is ready.

Main work was to add:
1. `cloudbuild.yaml` -- this simply runs go tests
1. `cloudbuild_build.yaml` -- this builds a docker image and pushes it to GCR (for now); eventually it will push to dockerhub once we figure out how to authenticate against dockerhub in cloud builder environment
1. `cloudbuild_deploy.yaml` -- Releases built image to a Kubernetes cluster using helm
1. `cloudbuild_build_deploy.yaml` -- Combines the build and deploy steps from the previous two files; this duplicates code from those files and edits to those files should be reflected here (and vice versa); side-effect of doc.ai deployment process.

**Comment by nkashy1** [comment]
Tested using:
```
gcloud builds submit --substitutions=SHORT_SHA=cloud-build-test,_VALUES_FILE="$VALUES_FILE",_CLOUDSDK_CORE_PROJECT=$GCP_PROJECT,_CLOUDSDK_COMPUTE_ZONE=$GCP_ZONE,_CLOUDSDK_CONTAINER_CLUSTER=$GKE_CLUSTER_NAME,_RELEASE_NAME=tio-models-test --config=cloudbuild_build_deploy.yaml ./
```

File: helm/values.yaml [hunk 1]
```diff
@@ -6,9 +6,9 @@ namespace: default
 
 deployment:
   replicas: 1
-  repository: tensorio/repository
+  repository: docai/tensorio-models
   tag: latest
```
Comment (brianbhsu) [inline helm/values.yaml]: In the future, this should be set to a specific version, otherwise there's a chance the `latest` build could be overwritten.

Comment (nkashy1) [reply helm/values.yaml]: :+1: This is just the sample `values.yaml` file. I am using git short sha for deployments into experimental, alpha, staging, production.

**Pull request merged**